### PR TITLE
Apple "universal binary framework" build script and project example

### PR DESCRIPTION
**Pull request by bigbrett**
# Description

Adds a script that shows how to build a "universal binary" library framework for all Apple devices that can be easily imported into a a multi-platform Xcode project. It also includes an example swiftUI application that calls a C driver function wolfCrypt tests and runs a simple TLS example

Gory details and demo app instructions can be found in the `README`

I added this to the IDE directory, but let me know if it should instead belong to another place or in another repo

### CODE REVIEW

**Review by dgarske — CHANGES_REQUESTED** [review]

**Comment by bigbrett** [comment]
@dgarske added license header to the build script and swift files too, let me know if that is excessive and I can remove.

File: IDE/apple-universal/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+`wolfssl-multiplatform` is an xcode project containing a simple swiftUI "hello world" app that has been modified to run the wolfCrypt tests and establish a TLS connection to `www.wolfssl.com` on startup. It also provides an example for basic Swift/C interoperability using a "bridging header". When the app launches, the swiftUI initialization handler calls a C test driver function, which is responsible for running the wolfSSL examples. An overview of the additional files is as follows:
+
+```
+.
+└── wolfssl-multiplatform
+    ├── wolfssl-multiplatform
+    │   ├── ContentView.swift # <-- boilerplate swiftUI modified to call wolfSSL test driver on UI init
+    │   ├── wolfssl_multiplatformApp.swift  # <-- basic swift hello world
+
+    │   ├── simple_client_example.c # <-- Simple TLS example that conects to wolfssl.com
```
Comment (dgarske) [inline IDE/apple-universal/README.md:25]: Typo `conects` -> `connects`.

File: IDE/apple-universal/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
+# Overview
+This example shows how to build a wolfSSL static library for Apple targets on all architectures using GNU autotools/`configure` and demonstrates how to create a [universal binary framework]() suitable for use in an Xcode project. It also provides a demo Xcode project using the wolfSSL framework in a simple multiplatform app.
```
Comment (dgarske) [inline IDE/apple-universal/README.md:2]: Please document what version of Xcode this is tested with... My 13.2.1 could not open the wolfssl-multiplatform.xcodeproj (too old).

File: IDE/apple-universal/build-wolfssl-framework.sh [hunk 3]
```diff
@@ -0,0 +1,295 @@
+#!/bin/bash
```
Comment (dgarske) [inline IDE/apple-universal/build-wolfssl-framework.sh:1]: Great script! Thank you

File: IDE/apple-universal/wolfssl-multiplatform/wolfssl-multiplatform/simple_client_example.c [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+    /* Create a new SSL context */
+    ctx = wolfSSL_CTX_new(wolfTLSv1_3_client_method());
+    if (ctx == NULL) {
+        printf("Unable to create SSL context.\n");
+        close(sockfd);
+        return 1;
+    }
+    
+    /* Load CA certificate into WOLFSSL_CTX
```
Comment (dgarske) [inline IDE/apple-universal/wolfssl-multiplatform/wolfssl-multiplatform/simple_client_example.c:86]: Line length cleanups. 80 chars max. Thanks

**Review by dgarske — CHANGES_REQUESTED** [review]

**Review by dgarske — DISMISSED** [review]
Over to @JacobBarthelmeh

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]
would be nice to have in the dist bundle for users, please update a include.am file for `make dist`

**Comment by bigbrett** [comment]
> would be nice to have in the dist bundle for users, please update a include.am file for `make dist`

@JacobBarthelmeh done

**Review by JacobBarthelmeh — APPROVED** [review]
Re-ran the Jenkins tests, they look ok just the status did not get updated.

Thanks Brett!

**Comment by bigbrett** [comment]
Added support for passing additional options to configure on the command line

**Review by JacobBarthelmeh — APPROVED** [review]

**Comment by JacobBarthelmeh** [comment]
> Added support for passing additional options to configure on the command line

Tested running the script with an added --enable-<option> and it picked it up. Thanks Brett!

**Pull request merged**